docs/troubleshooting-guide/sql-error-message.md

```diff
@@ -0,0 +1,199 @@
+   - You can also directly query the information schema in your database if you have permission to access it.
+
+2. Review the **function** section of the [reference guide][sql-reference-guide] for your SQL dialect.
+
+   - Confirm that the function exists for your SQL dialect.
+   - Review the data type(s) that are accepted by your function.
+
+3. If the field type of your column does not match the expected data type of your function:
+
+   - Cast your column to the correct data type in your SQL query.
+   - If you’re a Metabase admin, you can also [cast data types from the Data model page][how-to-cast-data-type].
+
+4. If you no longer have an error message, but your query results are incorrect, go to [Troubleshooting SQL logic][troubleshoot-sql-logic].
+5. If you're still stuck, search or ask the community on [Discourse][discourse].
+
+**Causes:**
+
+- SQL functions are designed to work on specific data types in your database.
+- For example, the [`DATE_TRUNC` function in PostgresSQL](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC) works on columns with `date`, `timestamp`, and `time` typed data in a Postgres database. If you try to use the `DATE_TRUNC` function on a column with a `string` data type in your database, it won’t work.
+- Note that Metabase [field types][field-type-def] are not one-to-one with the data types in your database. In this case, the field type gives you enough information about the column data type to troubleshoot the error.
+
+**Further reading:**
+
+- [How Metabase executes SQL queries][how-metabase-executes-queries].
+- [Field types documentation][field-type-doc].
+
+## I don’t know which line of my SQL query is failing.
+
+If your SQL query contains [SQL variables][sql-variable-def] that look like `{% raw %}{{ variable }}{% endraw %}`, go to [Troubleshooting SQL variables][troubleshoot-sql-variables] first.
+
+Once you find the line that is failing in your SQL query, go to the [Debugging SQL][sql-debugging] steps.
+
+### Reading your SQL error message
+
+- Does your error message tell you the line or character position?
+- Does your error message include a table or column name? If the table or column name appears more than once in your query, [reduce the size of your query][how-to-reduce-sql-query-size].
+- Does your error message mention a SQL clause?
+
+### Reducing the size of a SQL query
+
+- If your query uses subqueries (nested queries), run each subquery separately. Start with the inner subqueries and work your way out.
+
+- If your query uses CTEs, run each CTE separately. Start with your base CTE and work your way down the query.
+
+- If your query uses any SQL variables that point to Metabase models, run each model separately. Go to the model by opening the variables panel, or enter the model ID number from the variable in the Metabase search bar.
```

Comment:
I tested this in postgres and redshift (may be all SQL dialects?), if you're using another question and possibly models as a base table, just be extra careful about having lingering comments at the end of the script. 

Question 123
> select plan, created_at
> from dbt_models.self_service_accounts
> limit 10
> -- this comment causes a problem when using the question as a base table.

> select * from {{#123}} t

It will throw "ERROR: syntax error at end of input Position: X"